tests/fuzz/v1alpha3_fuzzer.go

```diff
@@ -0,0 +1,57 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint: golint
+package fuzz
+
+import (
+	"testing"
+
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/networking/core/v1alpha3"
+	"istio.io/istio/pilot/test/xdstest"
+)
+
+func init() {
+	testing.Init()
+}
+
+func FuzzValidateClusters(data []byte) int {
+	t := &testing.T{}
+	proxy := model.Proxy{}
+	f := fuzz.NewConsumer(data)
+	err := f.GenerateStruct(&proxy)
+	if err != nil {
+		return 0
+	}
+	to := v1alpha3.TestOptions{}
+	cg := v1alpha3.NewConfigGenTest(t, to)
+	clusters := cg.Clusters(cg.SetupProxy(&proxy))
```

Comment:
I think this will always generate empty response?

Reply:
My observation based on local tests is that `len(clusters)` is not `0` about 15% of the time.

Reply:
I think when its non-zero its also probably not getting meaningful config generation. I would guess we hit https://github.com/istio/istio/blob/fb62ce54058a1749ab747d5dd2780288525d1b76/pilot/pkg/networking/core/v1alpha3/cluster.go#L89 which is just building a static cluster that never changes. The real coverage would come in at https://github.com/istio/istio/blob/fb62ce54058a1749ab747d5dd2780288525d1b76/pilot/pkg/networking/core/v1alpha3/cluster.go#L163 which would require the push context to be non-empty to ever enter the loop

Reply:
I see. You are correct that the fuzzer does not make into [this loop](https://github.com/istio/istio/blob/19fb530b0c6176404fdba48de8d104f76d81cdea/pilot/pkg/networking/core/v1alpha3/cluster.go#L163)  on shorter runs. 

OSS-fuzz provides tooling that is great for tracking this, so what do you think about not having a branch in the fuzzer for the cluster validation and improve coverage once we get feedback from OSS-fuzz based on a few hundreds of hours of fuzzing? The current version reflects this suggestion.

Reply:
Ahh, I was missing that we fuzz TestOptions which can generate Services. I thought we only fuzzed Proxy which would never enter. SGTM then.

Maybe if the fuzzer doesn't get much overage we can add some seed info